src/main/java/taiga/util/pbChange/PBChangeCalculator.java

```diff
@@ -0,0 +1,49 @@
+package taiga.util.pbChange;
+
+import java.util.ArrayList;
+import java.util.List;
+import taiga.model.query.epic.EpicDetail;
+import taiga.model.query.history.History;
+import taiga.model.query.sprint.Sprint;
+import taiga.model.query.taskhistory.ItemHistory;
+import taiga.model.query.userstories.UserStoryInterface;
+
+public class PBChangeCalculator {
```

Comment:
The calculator should likely do the calculation itself, rather than delegating the work to another class. The two methods could likely be refactored to only include the sprint, from which it can derive all stories from the project (using the user stories API) and then proceed with filtering as normal.

Reply:
Removed any leftover history items, methods left unchanged to not violate SRP.